ApiReview/API.Polly.Core/NoDocs/Polly/Outcome.cs

```diff
@@ -0,0 +1,14 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Threading.Tasks;
+
+namespace Polly;
+
+public static class Outcome
+{
+    public static Outcome<TResult> FromResult<TResult>(TResult? value);
+    public static ValueTask<Outcome<TResult>> FromResultAsTask<TResult>(TResult value);
```

Comment:
while I find this helper useful, I believe if replace it with an extension for Outcome, it will enable more use-cases

`var task = Outcome.FromResult("dummy").AsTask()`

Reply:
See #1328 where we discuss this API (and `AsValueTask` as well). 

The following comment exaplains why we stuck with `FromResultAsTask`:

https://github.com/App-vNext/Polly/issues/1328#issuecomment-1598257460

Reply:
Consider renaming to `FromResultAsValueTask` to make clear what it returns